src/PKSim.Core/Services/ExpressionProfileUpdater.cs

```diff
@@ -150,6 +153,15 @@ public void SynchroniseSimulationSubjectWithExpressionProfile(ISimulationSubject
          var (sourceMolecule, sourceIndividual) = expressionProfile;
          // ExpressionProfile => SimulationSubject, we want to make sure that the parameters in simulation subject are linked to their expression profile origin parameters
          synchronizeExpressionProfiles(sourceMolecule, sourceIndividual, moleculeInIndividual, simulationSubject, updateParameterOriginId: true);
+
+         //Once the synchronization was performed, apply changes to simulation subject molecules based on disease state
+         updateMoleculeParametersForDiseaseState(simulationSubject, moleculeInIndividual);
+      }
+
+      private void updateMoleculeParametersForDiseaseState(ISimulationSubject simulationSubject, IndividualMolecule moleculeInIndividual)
+      {
```

Comment:
After everything has been updated, we apply the change again. This is a fairly straight forward implementaiton. We might want to do something more "clever" later on if needed.